### PR TITLE
Fix existence checks of packages lists for "Additional Packages" feature

### PR DESCRIPTION
**Pull request by thutterer**
This is a follow-up pull request to #420 
Refactoring introduced a bug that slipped through my tests.

The way I checked the existence of additional packages entries at the end raised an `undefined method '[]' for nil:NilClass` as long `:packages` did not exist yet.

### CODE REVIEW

File: chef/cookbooks/provisioner/recipes/setup_base_images.rb [hunk 1]
```diff
@@ -424,7 +424,11 @@
     Provisioner::Repositories.inspect_repos(node)
     repos = Provisioner::Repositories.get_repos(node, "suse", target_platform_version)
 
-    packages = node[:provisioner][:packages][os] || []
+    if node[:provisioner][:packages]
+      packages = node[:provisioner][:packages][os] || []
+    else
+      packages = []
+    end
```
Comment (rsalevsky) [inline chef/cookbooks/provisioner/recipes/setup_base_images.rb]: I prefer the following style but I'am also fine with leaving it as it is.
```
packages = []
if node[:provisioner][:packages]
  packages = node[:provisioner][:packages][os] || []
end
```

File: chef/cookbooks/provisioner/recipes/update_nodes.rb [hunk 2]
```diff
@@ -232,7 +232,11 @@
             end
           end
 
-          packages = node[:provisioner][:packages][os] || []
+          if node[:provisioner][:packages]
+            packages = node[:provisioner][:packages][os] || []
+          else
+            packages = []
+          end
```
Comment (rsalevsky) [inline chef/cookbooks/provisioner/recipes/update_nodes.rb]: Same again.

**Comment by rsalevsky** [comment]
Beside my small nitpick I'm fine, crowbar installation works again. :+1:

File: chef/cookbooks/provisioner/recipes/setup_base_images.rb (hunk 1 above)
Comment (thutterer) [inline chef/cookbooks/provisioner/recipes/setup_base_images.rb]: Actually I had your version before the evil rewrite. But I tend to favor my current style just for readability reasons. `If`/`else` seems to reflect the logic better than overwriting a set value in a simple `if`. But I guess this is a matter of taste and I'm happy to change it if someone else outvotes me.

**Comment by thutterer** [comment]
@rsalevsky I updated according to your suggestion.

**Comment by tboerger** [comment]
But you made a migration that created the packages key? so why do we need all the checks?

File: chef/cookbooks/provisioner/recipes/setup_base_images.rb (hunk 1 above)
Comment (tboerger) [inline chef/cookbooks/provisioner/recipes/setup_base_images.rb]: And i would prefer

```
packages = if node[:provisioner][:packages]
  node[:provisioner][:packages][os] || []
else
  []
end
```

**Comment by rsalevsky** [comment]
The packages are not required and we have no default values for them. So they are missing in an new environment.

File: chef/cookbooks/provisioner/recipes/setup_base_images.rb [hunk 3]
```diff
@@ -424,7 +424,10 @@
     Provisioner::Repositories.inspect_repos(node)
     repos = Provisioner::Repositories.get_repos(node, "suse", target_platform_version)
 
-    packages = node[:provisioner][:packages][os] || []
+    packages = []
+    if node[:provisioner][:packages]
+      packages = node[:provisioner][:packages][os] || []
+    end
```
Comment (thutterer) [inline chef/cookbooks/provisioner/recipes/setup_base_images.rb]: >And i would prefer
```
packages = if node[:provisioner][:packages]
  node[:provisioner][:packages][os] || []
else
  []
end
```

No strong opinion on my side. Others?

**Comment by thutterer** [comment]
In an offline discussion @tboerger suggested to loose the code changes in this PR and instead make `:packages` a required attribute and set it's default value to `[ ]`.

**Comment by dirkmueller** [comment]
and how would that make [packages][os] a required attribute ? that change would only solve one level of if's

**Comment by thutterer** [comment]
I think #426 is a cleaner way to fix this. That way, `packages` attribute is always in the JSON and we can depend on it in the code without multiple conditions.

**Comment by rsalevsky** [comment]
#426 is merged

**Pull request closed without merging**